source/core/constants.ts

```diff
@@ -1,18 +1,23 @@
+import type { Expect, Equal } from '@type-challenges/utils';
 import {HTTPMethod} from '../types/options.js';
 
 export const supportsAbortController = typeof globalThis.AbortController === 'function';
 export const supportsStreams = typeof globalThis.ReadableStream === 'function';
 export const supportsFormData = typeof globalThis.FormData === 'function';
 
-export const requestMethods: readonly HTTPMethod[] = ['get', 'post', 'put', 'patch', 'head', 'delete'];
+export const requestMethods = ['get', 'post', 'put', 'patch', 'head', 'delete'] as const;
+
+type cases = [
+	Expect<Equal<typeof requestMethods[number], HTTPMethod>>,
+]
```

Comment:
instead of declaring `requestMethods` as `readonly HTTPMethod[]`, you make it readonly as is

and then check here in `cases` that values are exactly the same